airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/SnowflakeS3StagingDestination.java

```diff
@@ -0,0 +1,108 @@
+import io.airbyte.integrations.destination.jdbc.SqlOperations;
+import io.airbyte.integrations.destination.s3.S3DestinationConfig;
+import io.airbyte.integrations.destination.staging.StagingConsumerFactory;
+import io.airbyte.protocol.models.AirbyteConnectionStatus;
+import io.airbyte.protocol.models.AirbyteMessage;
+import io.airbyte.protocol.models.ConfiguredAirbyteCatalog;
+import java.util.Collections;
+import java.util.Map;
+import java.util.UUID;
+import java.util.function.Consumer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SnowflakeS3StagingDestination extends AbstractJdbcDestination implements Destination {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(SnowflakeS3StagingDestination.class);
+
+  public SnowflakeS3StagingDestination() {
+    this(new SnowflakeSQLNameTransformer());
+  }
+
+  public SnowflakeS3StagingDestination(final SnowflakeSQLNameTransformer nameTransformer) {
+    super("", nameTransformer, new SnowflakeSqlOperations());
+  }
+
+  @Override
+  public AirbyteConnectionStatus check(final JsonNode config) {
+    final S3DestinationConfig s3Config = getS3DestinationConfig(config);
+    final NamingConventionTransformer nameTransformer = getNamingResolver();
+    final SnowflakeS3StagingSqlOperations SnowflakeS3StagingSqlOperations = new SnowflakeS3StagingSqlOperations(nameTransformer, s3Config.getS3Client(), s3Config);
+    try (final JdbcDatabase database = getDatabase(config)) {
+      final String outputSchema = super.getNamingResolver().getIdentifier(config.get("schema").asText());
+      AirbyteSentry.executeWithTracing("CreateAndDropTable",
+          () -> attemptSQLCreateAndDropTableOperations(outputSchema, database, nameTransformer, SnowflakeS3StagingSqlOperations));
+      AirbyteSentry.executeWithTracing("CreateAndDropStage",
+          () -> attemptSQLCreateAndDropStages(outputSchema, database, nameTransformer, SnowflakeS3StagingSqlOperations));
+      return new AirbyteConnectionStatus().withStatus(AirbyteConnectionStatus.Status.SUCCEEDED);
+    } catch (final Exception e) {
+      LOGGER.error("Exception while checking connection: ", e);
+      return new AirbyteConnectionStatus()
+          .withStatus(AirbyteConnectionStatus.Status.FAILED)
+          .withMessage("Could not connect with provided configuration. \n" + e.getMessage());
+    }
+  }
+
+  private static void attemptSQLCreateAndDropStages(final String outputSchema,
+                                                    final JdbcDatabase database,
+                                                    final NamingConventionTransformer namingResolver,
+                                                    final SnowflakeS3StagingSqlOperations sqlOperations)
+      throws Exception {
+
+    // verify we have permissions to create/drop stage
+    final String outputTableName = namingResolver.getIdentifier("_airbyte_connection_test_" + UUID.randomUUID().toString().replaceAll("-", ""));
```

Comment:
since `NamingConventionTransformer` is a `SnowflakeSQLNameTransformer` which is specific to Snowflake, could we hide the `replaceAll` call inside the `getIdentifier` method?

Reply:
the identifiers in snowflake are authorized to have `-` in them, I don't know why this code is doing this replace on the table name, it seems unnecessary... but since it's only for a test of connectivity, it doesn't matter much, let me remove it

Reply:
it's probably unecessary to call replaceAll here since the getIdentifier should already get rid of '-', removing it